olx_parser.py

```diff
@@ -28,97 +28,132 @@
         'ноября': 'November',
         'декабря': 'December'}
 
+def translate_month_to_en(ru_month_name: str) -> str:
+    # Функция перевода месяца с RU на EN
+    return month_mapping[ru_month_name]
 
-def get_product(url, count = 100):
+
+'''Сложно воспринимать такую длинную функцию и с такой вложенностью циклов. Давай разобьём на несколько:
+
+1. парсинг категории parser_category
+2. парсинг одной страницы в категории parser_page
+3. парсинг одного объявления на странице parser_adt'''
+
+def parser_page(url, count = 100):
+    count_error = 1
     title_begin = None
     title_str = None
-    for page in range(1,count):
+    for page in range(1,26):
         title_begin = title_str
         try:
             response = requests.get(f'{url}{page}', timeout=5)
             response.raise_for_status()
             soup = BeautifulSoup(response.text, 'lxml')
             title_str = soup.find('title').text
             if title_begin == title_str:
-                break
+                break  
             data = soup.find_all('div', class_='css-1sw7q4x')
-            for link in data:
-                card_url = link.find('a').get('href')
-                response_product = requests.get('https://www.olx.kz' + card_url, timeout=5)
-                response_product.raise_for_status()                
-                soup_product = BeautifulSoup(response_product.text, 'lxml')
-                soup_product_number_id = soup_product.find('span', class_='css-12hdxwj er34gjf0').text                 
-                soup_product_name_salesman = soup_product.find('h4', class_='css-1lcz6o7 er34gjf0').text
-
-                soup_product_date_registered = soup_product.find('div', class_='css-16h6te1 er34gjf0').text.split(' ')
-                date_registered = format_date(soup_product_date_registered)
-                
-
-                soup_product_date_of_last_visit = soup_product.find('span', class_='css-1t0qnkx').text.split(' ')
-                date_of_last_visit = format_date(soup_product_date_of_last_visit)
-                
-                soup_product_date_posted = soup_product.find('span', class_='css-19yf5ek').text.split(' ')
-                date_posted = format_date(soup_product_date_posted)
-                
-                soup_product_jpg = soup_product.find('img').get('src')                  
-                soup_product_title = soup_product.find('h4', class_='css-1juynto').text
-                soup_product_text = soup_product.find('div', class_='css-1t507yq er34gjf0').text
-        
+            parser_adt(data, count_error)
```

Comment:
Вот в этом месте хорошо зайдёт yield
```suggestion
            parsed_adt = parser_adt(data, count_error)
            yield parsed_adt
```
Тогда функция parse_page станет генератором и можно будет делать типа такого
```python
for adt_dict in parse_page('ololo.com', 100):
    db.insert_adt(adt_dict)
```